File: WhatIfBudget/WhatIfBudget.Data/Models/Budget.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WhatIfBudget.Data.Models
{
    public class Budget : BaseEntity
    {
        public Guid UserId { get; set; }
        public string Name { get; set; } = String.Empty;
        public int SavingGoalId { get; set; }
        public int DebtGoalId { get; set; }
        public int MortgageGoalId { get; set; }
        public int InvestmentGoalId { get; set; }

        public virtual InvestmentGoal? InvestmentGoal { get; set; }
        public virtual SavingGoal? SavingGoal { get; set; }
        public virtual MortgageGoal? MortgageGoal { get; set; }
        public virtual DebtGoal? DebtGoal { get; set; }
    }
}

File: WhatIfBudget/WhatIfBudget.App/ClientApp/src/app/shared/models/budget.ts
import { KeyValue } from "@angular/common";

export class Budget {
  id: number | undefined;
  name: string | undefined;
  savingGoalId: number | undefined;
  debtGoalId: number | undefined;
  mortgageGoalId: number | undefined;
  investmentGoalId: number | undefined;
}

export class AdditionalContributions {
  debtGoal: number | undefined;
  mortgageGoal: number | undefined;
  savingGoal: number | undefined;
  investmentGoal: number | undefined;
}

export class NetValue {
  balance: KeyValue<number, number>[] = [];
  savingGoalMonth: number | undefined;
  debtGoalMonth: number | undefined;
  mortgageGoalMonth: number | undefined;
}
